errors.go

```diff
@@ -3,37 +3,58 @@ package yggdrasil
 import (
 	"fmt"
 	"net/http"
+	"strings"
+
+	pb "github.com/redhatinsights/yggdrasil/protocol"
 )
 
 // ErrInvalidContentType indicates an unsupported "collector" value was given
 // in the upload request.
-var ErrInvalidContentType = &APIResponseError{
+var ErrInvalidContentType = &APIresponse{
 	Code: http.StatusUnsupportedMediaType,
-	Body: "Content type of payload is unsupported",
+	Body: []byte("Content type of payload is unsupported"),
 }
 
 // ErrPayloadTooLarge indicates an upload request body exceeded the size limit.
-var ErrPayloadTooLarge = &APIResponseError{
+var ErrPayloadTooLarge = &APIresponse{
 	Code: http.StatusRequestEntityTooLarge,
-	Body: "Payload too large",
+	Body: []byte("Payload too large"),
 }
 
 // ErrUnauthorized indicates an upload request without an Authentication header.
-var ErrUnauthorized = &APIResponseError{
+var ErrUnauthorized = &APIresponse{
 	Code: http.StatusUnauthorized,
-	Body: "Authentication missing from request",
+	Body: []byte("Authentication missing from request"),
+}
+
+// An APIresponse represents an unexpected response from an HTTP method call.
+type APIresponse struct {
```

Comment:
The type looks good but I am not sure whether it should be in this file. It feels correct to have it in `http.go`. What do you think?

Reply:
yup, to be moved.